### PR TITLE
[CET-2856] Add rule filters

### PR DESCRIPTION
**Pull request by eyalfoni**
## Change description

> Description here

This PR exposes rule filters on the Scorecard details page alongside the rule. Rule filters for CQL and group inclusion/exclusion are accounted for.

Screen Shot 2023-05-18 at 11 08 26 AM
Screen Shot 2023-05-18 at 11 06 07 AM

See Scorecard detail page rule will be shown as the following if it is filtered out:

Screen Shot 2023-05-17 at 3 23 31 PM

## Type of change

- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)
- [ ] Improvement (improves codebase without changing functionality)

## Related issues

> Fix [#1]()

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: src/components/Scorecards/ScorecardDetailsPage/ScorecardRulesCard/ScorecardRuleRow.tsx [hunk 1]
```diff
@@ -67,8 +73,37 @@ export const ScorecardRuleRow = ({ rule }: ScorecardRuleRowProps) => {
...
         </Collapse>
       </Grid>
-      <Grid item lg={2}>
-        {hasWeight && <b>{rule.weight}</b>}
+      <Grid item xs={4}>
+        {hasWeight && (
+          <Typography className={classes.rule}>{rule.weight}</Typography>
+        )}
+      </Grid>
+      <Grid item xs={4}>
```
Comment (cristinabuenahora) [inline src/components/Scorecards/ScorecardDetailsPage/ScorecardRulesCard/ScorecardRuleRow.tsx:81]: Not a huge deal but I wonder if we should but this in the collapsible with the desc. It looks busy as is and could look busier if there's lots of groups / a long query.

Comment (eyalfoni) [reply src/components/Scorecards/ScorecardDetailsPage/ScorecardRulesCard/ScorecardRuleRow.tsx:81]: Good point, do you mean only the filter in a collapsible and leave desc/weight the same?

Comment (cristinabuenahora) [reply src/components/Scorecards/ScorecardDetailsPage/ScorecardRulesCard/ScorecardRuleRow.tsx:81]: Desc is already in collapsible right? So put it below that in the same collapsible

Comment (eyalfoni) [reply src/components/Scorecards/ScorecardDetailsPage/ScorecardRulesCard/ScorecardRuleRow.tsx]: This is way better! See new screenshots in PR description.

File: src/api/types.ts [hunk 2]
```diff
@@ -170,8 +178,8 @@ export interface ApplicableRuleOutcome extends RuleOutcomeBase {
 
 export interface NotApplicableRuleOutcome extends RuleOutcomeBase {
   endDate?: string;
-  requestedDate: string;
-  approvedDate: string;
+  requestedDate?: string;
+  approvedDate?: string;
```
Comment (FoodProduct) [inline src/api/types.ts:182]: When would these be null?

Comment (eyalfoni) [reply src/api/types.ts:182]: We're using the not applicable outcome for rule filters as well as rule exemptions, so these don't have any request/approval process and would be null in this case

File: src/components/Scorecards/ScorecardDetailsPage/ScorecardRulesCard/ScorecardRuleRow.tsx [hunk 3]
```diff
@@ -64,6 +70,35 @@ export const ScorecardRuleRow = ({ rule }: ScorecardRuleRowProps) => {
                 {rule.expression}
               </MetadataItem>
             )}
+            {rule.filter?.query && (
+              <MetadataItem gridSizes={{ xs: 12 }} label="Filter">
+                {rule.filter.query}
+              </MetadataItem>
+            )}
+            {rule.filter?.includedTags &&
+              rule.filter?.includedTags.length !== 0 && (
```
[the remark this answers is not in the document]
Comment (eyalfoni) [reply src/components/Scorecards/ScorecardDetailsPage/ScorecardRulesCard/ScorecardRuleRow.tsx:79]: seeing `TS2532: Object is possibly 'undefined'.` when i do this

although the following works:
```
rule.filter?.includedTags && 
                !isEmpty(rule.filter?.includedTags) && (
```

File: src/api/types.ts (hunk 2 above)
Comment (FoodProduct) [reply src/api/types.ts:182]: Ah, good to know!

File: src/components/Scorecards/ScorecardDetailsPage/ScorecardRulesCard/ScorecardRuleRow.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (FoodProduct) [reply src/components/Scorecards/ScorecardDetailsPage/ScorecardRulesCard/ScorecardRuleRow.tsx:79]: Ugh TS. Technically we could do `!isNil(x) && !isEmpty(x)` but then that doesn't benefit us very much 😿

**Review by FoodProduct — APPROVED** [review]
I didn't pull this down myself but the code and screenshots LGTM

**Pull request merged**